File: December 2024 Leetcode Solution/December-02.java
class Solution {
    public int isPrefixOfWord(String sentence, String searchWord) {
        int n = sentence.length(), m = searchWord.length();
        int word_index = 1, i = 0;
        
        while (i < n) {
            if (sentence.startsWith(searchWord, i) && (i == 0 || sentence.charAt(i - 1) == ' ')) {
                return word_index;
            }
            while (i < n && sentence.charAt(i) != ' ') {
                i++;
            }
            word_index++;
            i++; 
        }
        
        return -1;
    }
}

2)
class Solution {
    public int isPrefixOfWord(String sentence, String searchWord) {
        String[] words = sentence.split(" "); 
        for (int i = 0; i < words.length; i++) {
            if (words[i].startsWith(searchWord)) { 
                return i + 1; 
            }
        }
        return -1; 
    }
}

File: December 2024 Leetcode Solution/December-02.c
int isPrefixOfWord(char* sentence, char* searchWord) {
    int n = strlen(sentence), m = strlen(searchWord);
    int word_index = 1, i = 0;
    
    while (i < n) {
        if (strncmp(&sentence[i], searchWord, m) == 0 && (i == 0 || sentence[i - 1] == ' ')) {
            return word_index;
        }
        
        while (i < n && sentence[i] != ' ') {
            i++;
        }
        word_index++;
        i++; 
    }
    
    return -1;
}

2)
int isPrefixOfWord(char* sentence, char* searchWord) {
    int wordIndex = 1; 
    char* word = strtok(sentence, " "); 

    while (word != NULL) {
        if (strncmp(word, searchWord, strlen(searchWord)) == 0) {
            return wordIndex;
        }
        wordIndex++;
        word = strtok(NULL, " ");
    }

    return -1; 
}
